mypy/checker.py

```diff
@@ -3719,6 +3713,78 @@ def find_isinstance_check(self, node: Expression
         # Not a supported isinstance check
         return {}, {}
 
+    def narrow_given_equality(self,
+                              left_expr: Expression,
+                              left_type: Type,
+                              right_expr: Expression,
+                              right_type: Type,
+                              assume_identity: bool,
+                              ) -> Tuple[TypeMap, TypeMap]:
+        """Assuming that the given 'left' and 'right' exprs are equal to each other, try
+        producing TypeMaps refining the types of either the left or right exprs (or neither,
+        if we can't learn anything from the comparison).
+
+        For more details about what TypeMaps are, see the docstring in find_isinstance_check.
+
+        If 'assume_identity' is true, assume that this comparison was done using an
+        identity comparison (left_expr is right_expr), not just an equality comparison
+        (left_expr == right_expr). Identity checks are not overridable, so we can infer
+        more information in that case.
+        """
+
+        # For the sake of simplicity, we currently attempt inferring a more precise type
+        # for just one of the two variables.
```

Comment:
How important is this? If we actually do this, I think it is better to use the type of right one to restrict the type of left one.